### DISCUSSION TITLE
Error in account-baseline

### DISCUSSION THREAD

**Discussion by moos3** [gruntwork-io/knowledge-base#434] [Help] [answered]
lately I have been seeing this pop-up more and more when running terraform plan's and applies on the reference arch. our reference arch is tf pinned to 1.0.11 and latest version of terragrunt 0.37.1. 

```
INFO[0000] The stack at /Users/moos3/workspaces/boost/terraform/infrastructure/shared/_global/account-baseline will be processed in the following order for command plan:
Group 1
- Module /Users/moos3/workspaces/boost/terraform/infrastructure/shared/_global/account-baseline

╷
│ Warning: Argument is deprecated
│
│   with module.app_baseline.module.cloudtrail.module.bucket.module.private_bucket.aws_s3_bucket.bucket,
│   on .terraform/modules/app_baseline.cloudtrail/modules/private-s3-bucket/main.tf line 20, in resource "aws_s3_bucket" "bucket":
│   20: resource "aws_s3_bucket" "bucket" {
│
│ Use the aws_s3_bucket_cors_configuration resource instead
│
│ (and 442 more similar warnings elsewhere)
╵
╷
│ Error: Unrecognized remote plugin message:
│
│ This usually means that the plugin is either invalid or simply
│ needs to be recompiled to support the latest protocol.
│
│
╵
ERRO[0021] Module /Users/moos3/workspaces/boost/terraform/infrastructure/shared/_global/account-baseline has finished with an error: 1 error occurred:
	* exit status 1
  prefix=[/Users/moos3/workspaces/boost/terraform/infrastructure/shared/_global/account-baseline]
INFO[0021] ╷
│ Error: Unrecognized remote plugin message:
│
│ This usually means that the plugin is either invalid or simply
│ needs to be recompiled to support the latest protocol.
│
│
╵
ERRO[0021] 1 error occurred:
	* exit status 1
```

---

<ins datetime="2022-05-25T13:08:42Z">
  <p><a href="https://support.gruntwork.io/hc/requests/108659">Tracked in ticket #108659</a></p>
</ins>

**Comment by yorinasub17** [comment, marked as answer]
> This usually means that the plugin is either invalid or simply needs to be recompiled to support the latest protocol.

This usually means your provider binary is outdated and needs a refresh. You should be able to address this by removing the terragrunt cache folder (`.terragrunt-cache` in the folder where `terragrunt.hcl` exists) and then trying again so that a fresh provider binary is pulled down.

**Reply by moos3** [reply]
Yeah I have tried that multiple times, then it starts complaining about regions that I did not opt into.

**Reply by yorinasub17** [reply]
Ah gotcha. For the regions issue, we recently found a bug in the configuration for the multi-region setup in the Reference Architecture (see https://github.com/gruntwork-io/terraform-aws-service-catalog/issues/1456). You should be able to resolve that by applying the same changeset as what's done in https://github.com/gruntwork-io/terraform-aws-service-catalog/pull/1461 for each of the files in the `for-production` folder.
